orchagent/dash/dashtunnelorch.cpp

```diff
@@ -127,6 +132,11 @@ void DashTunnelOrch::doTask(ConsumerBase &consumer)
                 if (addTunnel(tunnel_name, ctxt))
                 {
                     it = consumer.m_toSync.erase(it);
+                    /*
+                     * Write result only when removing from consumer in pre-op
+                     * For other cases, this will be handled in post-op
+                     */
+                    writeResultToDB(dash_tunnel_result_table_, tunnel_name, result);
```

Comment:
We need to report a failure here if we're removing from the consumer after pre-op

Reply:
discussed offline, leaving for now until return value can capture more detail

Reply:
Added a TODO.